### DISCUSSION TITLE
Directus as storage / database backend instead of ?

### DISCUSSION THREAD

**Discussion by pwFoo** [directus/directus#4647] [Archive]
Hi,
I looked into Directus as headless cms to deliver data to clients / apps independent from (html) templates. So Directus at server side and a JavaSript client app directly use the api to get data to the client.

Is Directus fast enough to be used as database framework / orm / file storage for a php application compared to a database framework like [RedBeanPHP](https://redbeanphp.com/index.php) or [Medoo](https://medoo.in/). It's clear that a simple PHP PDO database layer or Medoo should be faster, but would Directus be a bottleneck used instead of such a database framework or should it be fast enough?

Would be interesting to know more about Directus performance and speed.

**Comment by rijkvanzanten** [comment]
> It's clear that a simple PHP PDO database layer or Medoo should be faster, but would Directus be a bottleneck used instead of such a database framework or should it be fast enough?

It's really hard to give a conclusive answer about that without knowing exactly what your use case is going to be.. There's a lot of variables that can impact the performance of your setup, including where Directus, the database, and your app are hosted. Generally speaking, Directus' overhead is pretty minimal.

There's also a third option you can consider, if straight-to-db performance is required:

Use Directus as an admin panel for your database, and then keep using RedBeanPHP / Medoo to read straight from the database in your app. That's hands down the most performant way to read the data. (However, you'll obviously won't be able to rely on Directus' authentication/authorization/filtering capabilities from within your app).

**Reply by pwFoo** [reply]
Thanks, no special use case in mind. I think small and not high performance critical app(s). 

The idea is to have the benefit of directus rest api instead of a db framework with additional custom rest api code.
Just to use directus the same solution for both types of applications. First test with few database  rows was fast. So I think directus could be that one tool for all my small use cases 👍 

Combine both if higher db performance is needed a good point! 

Thanks!

**Reply by rijkvanzanten** [reply]
> I think small and not high performance critical app(s).

In that case, Directus should be more than fine to handle what you're throwing at it 🙂
